trace_point.rb

```diff
@@ -168,14 +168,15 @@ def self.allow_reentry
   #                   #   trace is still enabled
   #
   # If a block is given, the trace will only be enabled within the scope of the
-  # block.
+  # block. If target and target_line are both nil, then target_thread will default
+  # to the current thread if a block is given.
```

Comment:
I think if previously you were explicitly specifying a target, we should assume you know what you are doing.  If you previously targeted a line without targeting a thread, I don't think we should force targeting the current thread.

Reply:
It's probably extremely rare to use `enable { ... }` + target/target_line, but I think in such cases the intuition is still it should only apply during the block call.
I think it is worth avoiding exceptions here for consistency and simplicity, I think the arguments on the issue basically apply to this case too, i.e., they apply to all usages of `enable { ... }`.

Reply:
I agree on the rarity, but I think the better principle is that automatically targeting the current thread is only done if no targets are specified.  We should assume users who specify targets know what they are doing.  Note that there is different behavior in both cases, it just depends on whether you draw the line.  You want thecondition to be block_given?, but I think a better condition is block_given? and no target specified.  My approach is also more backwards compatible.

Reply:
>  We should assume users who specify targets know what they are doing.

I wouldn't assume that, actually I think almost everyone who would have written:
`tp.enable(target: ...) { ... }` would have expected to only trigger `tp` if called on the current thread, while that block is being called.
And it might have accidentally worked as expected if there is no other thread calling that `target`.
But of course it's hard to know for sure.

@ko1 @mame Any opinion on whether the "on current thread" filter should be for all `enable { ... }`, or only if block & no target?

Reply:
with well written documentation, I think both okay for me.

Reply:
I consider the documentation I added well written :), so I plan to merge this unless @mame would prefer different behavior.